syntax = "proto3";

package user;

//The service we are going to expose
service UserProvider {
  // For Auth Service
  rpc GetUserByEmail (UserRequest) returns (UserResponse);
}

message UserRequest {
  string email = 1;
}

message UserResponse {
  string id = 1;
  string email = 2;
  string password_hash = 3;
  string role = 4;
  bool is_active = 5;
  bool found = 6;
}